Userland/Libraries/LibGL/SoftwareGLContext.h

```diff
@@ -78,6 +107,70 @@ class SoftwareGLContext : public GLContext {
     Clipper m_clipper;
 
     SoftwareRasterizer m_rasterizer;
+
+    struct Listing {
```

Comment:
Can we put this in its own file? Also this should probably be a class, I think structs are mainly reserved for POD types. Correct me if I'm wrong.

Reply:
Making it a class is just adding trouble, this has no particular behaviour, it's just a bunch of fields stuck together.
if we make it a class, it'll just be more code that does nothing.

Our [coding style](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#classes) seems to be saying "prefer classes when it has methods" :shrug:

Reply:
Oh you are right.. reviewing heavily templated code on a smartphone was probably not the best idea...

Reply:
Oh, sorry I forgot to answer this one:
> Can we put this in its own file?

Not the whole thing, it depends on the types of the methods being known (`&SoftwareGLContext::gl_foo`), but `SoftwareGLContext` depends on this struct being defined, so it basically has to go inside `SoftwareGLContext`.

I could move the utility types (`TupleTypeForArgumentListOf`, `ArgumentsFor` and `FunctionAndArgs`) to a separate file if you want though